ci: intermittent k8s 1.19 and 1.20 test failures

I saw this issue in the CI system, it was showing up in a k8s 1.20 test that was just added. I'm not sure what is the cause yet, but it would be nice to print more info than the status code as a first step and then debug further.

```
    @pytest.fixture(scope="function")
    def jupyter_user(api_request):
        """
        A temporary unique JupyterHub user
        """
        username = "testuser-" + str(uuid.uuid4())
        r = api_request.post("/users/" + username)
        assert r.status_code == 201
        yield username
        r = api_request.delete("/users/" + username)
>       assert r.status_code == 204
E       assert 400 == 204
E         +400
E         -204
```

https://github.com/jupyterhub/zero-to-jupyterhub-k8s/pull/1961/checks?check_run_id=1594410192

## Update

The failure is intermittent, about 50% of the time in our ci tests. I added some logging and such allowing the failure to be inspected [in this run](https://github.com/consideRatio/zero-to-jupyterhub-k8s/runs/1596599338?check_suite_focus=true).

The conclusion I draw is that there is some pod termination issues in k8s 1.20, and it sais...

```
    State:          Waiting
      Reason:       PodInitializing
    Last State:     Terminated
      Reason:       ContainerStatusUnknown
      Message:      The container could not be located when the pod was deleted.  The container used to be Running
      Exit Code:    137
      Started:      Mon, 01 Jan 0001 00:00:00 +0000
      Finished:     Mon, 01 Jan 0001 00:00:00 +0000
```

To me, this is very confusing, but googling for k8s 1.20 and this message, I ended up in a k8s 1.20 issue. For reference, we are using k3s 1.20.0+k3s2 currently.

https://github.com/kubernetes/kubernetes/issues/97288

--- a/.github/workflows/test-chart.yaml
+++ b/.github/workflows/test-chart.yaml
@@ -100,10 +100,6 @@ jobs:
         include:
           - k3s-channel: v1.20
             test: install
-            # FIXME: stop accepting failure when 1.20.3 is available for k3s and
-            #        this issue is resolved:
-            #        https://github.com/jupyterhub/zero-to-jupyterhub-k8s/issues/1962
-            accept-failure: true
           - k3s-channel: v1.19
             test: install
           - k3s-channel: v1.18

